newIDE/app/src/ProjectManager/PublicGamePropertiesDialog.js

```diff
@@ -33,6 +36,24 @@ function applyPublicPropertiesToProject(
   return displayProjectErrorsBox(t, getProjectPropertiesErrors(t, project));
 }
 
+/**
+ * Public game properties that are not stored in a project file.
+ */
+type PublicGameOnlyProperties = {|
+  playWithKeyboard: boolean,
+  playWithGamepad: boolean,
+  playWithMobile: boolean,
+|};
+
+function applyPublicPropertiesToGame(
+  publicGame: PublicGame,
+  newProperties: PublicGameOnlyProperties
+) {
+  publicGame.playWithKeyboard = newProperties.playWithKeyboard;
```

Comment:
It's a bit against the "React way" to mutate things in place, as in the future other components relying on the publicGame could not see the object was changed and don't re-render.
Instead, I would instanciate a new publicGame with the updated attributes:

`const updatedPublicGame = { ...publicGame, playWithKeyboard: ..., ...}`

And I would pass it to the parent inside the `onApply`.
Finally, in the `onApply={...}` I would call `updateGameFromProject` by passing it the updatedPublicGame:
`updateGameFromProjectAndPublicGame(updatedPublicGame)`

(for the project, it's a bit different because objects from libGD.js *are* mutated in place, and this is done almost everywhere).

And then it's the parent responsibility to update its public game, which it will do already because the game is updated.

---

OR we add the properties in the project now (even if not displayed in the project settings dialog).

Reply:
I'll go for the 2nd option as it will avoid to revert it in a few days.

Reply:
Good decision! :)